pep-0630.rst

```diff
@@ -321,10 +328,26 @@ the Python level: for example, you can't set ``str.myattribute = 123``.
    that shares any Python objects across interpreters implicitly depends
    on CPython's current, process-wide GIL.
 
-An alternative to static types is *heap-allocated types*, or heap types
+Because they are immutable and process-global, static types cannot access
+“their” module state.
+If any method of such a type requires access to module state,
+the type must be converted to a *heap-allocated type*, or *heap type*
 for short. These correspond more closely to classes created by Python’s
 ``class`` statement.
 
+For new modules, using heap types by default is a good rule of thumb.
+
+Existing classes can be converted to heap ones, but note that
+the heap type API was not designed for “lossless” conversion
+from static types. Unlike static types, heap type objects are mutable.
```

Comment:
Heap type objects are _mutable by default_. With the current wording it sounds like they are always mutable; that is not true. I strongly suggest you reconsider your wording, in order to avoid misunderstandings. Explicit is better than implicit.

```suggestion
from static types. Unlike static types, heap type objects are mutable by default.
Heap types can be made immutable by using the flag ``Py_TPFLAGS_IMMUTABLETYPE``.
```

Reply:
Thanks! Somehow I forgot about that 3.10 addition of Py_TPFLAGS_IMMUTABLETYPE. 
Adding "by default" makes the PEP technically correct, though it'll be a while before common libraries can drop 3.9 and use immutable heap types.
Luckily, having types immutable is a lot more important in stdlib than elsewhere.